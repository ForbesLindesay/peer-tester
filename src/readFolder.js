import lsr from 'lsr';
import {readFile} from 'then-fs';

async function readFolder(dirname) {
  const entries = await lsr(dirname);
  return Promise.all(
    entries.map(async (entry) => {
      if (entry.isFile()) {
        return {
          type: 'file',
          path: entry.path,
          content: await readFile(entry.fullPath),
        };
      } else {
        return {
          type: 'directory',
          path: entry.path,
        };
      }
    }),
  );
}

export default readFolder;
