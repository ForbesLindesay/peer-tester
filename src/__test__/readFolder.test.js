import readFolder from '../readFolder';

test('readFolder(exampleDir)', async () => {
  const files = await readFolder(__dirname + '/../../example');
  expect(files).toMatchSnapshot();
});
