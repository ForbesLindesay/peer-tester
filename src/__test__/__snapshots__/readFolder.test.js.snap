// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readFolder(exampleDir) 1`] = `
Array [
  Object {
    "content": Buffer("{
  \\"name\\": \\"example\\",
  \\"private\\": true,
  \\"peerDependencies\\": {
    \\"promise\\": \\"^6.0.0\\"
  },
  \\"scripts\\": {
    \\"test\\": \\"node test\\"
  }
}
"),
    "path": "./package.json",
    "type": "file",
  },
  Object {
    "path": "./test",
    "type": "directory",
  },
  Object {
    "content": Buffer("console.log('promise version v' + require('promise/package.json').version);
if (require('promise/package.json').version === '6.0.1') {
  console.log('Simulating faield test for v6.0.1');
  process.exit(1);
}
"),
    "path": "./test/index.js",
    "type": "file",
  },
]
`;
